src/main/java/sirius/web/services/PublicServices.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.web.services;
+
+import sirius.kernel.Sirius;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.settings.Extension;
+import sirius.web.controller.Routed;
+import sirius.web.http.WebServer;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Lists all known {@link PublicApiInfo public APIs}.
+ * <p>
+ * Public APIs are discovered by the web-server when building the routes and recognizing the
+ * {@link PublicService} annotations.
+ */
+@Register(classes = PublicServices.class)
+public class PublicServices {
+
+    private final List<PublicApiInfo> apis = new ArrayList<>();
+
+    /**
+     * Lists all known public APIs.
+     *
+     * @return the list of all known public APIs
+     */
+    public List<PublicApiInfo> getApis() {
+        synchronized (apis) {
+            return new ArrayList<>(apis);
```

Comment:
unmodifiableList?

Reply:
We need to copy because of threading issues - the copy doesn't need to be unmodifyable as it is unused elsewhere..